kernel/src/events/subject.rs

```diff
@@ -83,7 +90,7 @@ impl<E: Events, F: EventsFilter<E>> Subject<E, F> {
         observers.retain(|observer, filter| {
             if let Some(observer) = observer.upgrade() {
                 if filter.filter(events) {
-                    // XXX: Mind the performance impact when there comes many active observers
+                    observer.on_events(events);
```

Comment:
Since the `on_events` has been moved forward, we don't need to maintain the `active_observers` vector?